addons/html_editor/static/src/core/selection_plugin.js

```diff
@@ -475,16 +499,12 @@ export class SelectionPlugin extends Plugin {
         const anchor = { node: selection.anchorNode, offset: selection.anchorOffset };
         const focus = { node: selection.focusNode, offset: selection.focusOffset };
 
-        let documentSelectionToRestore;
-        if (!selectionData.documentSelectionIsInEditable) {
-            documentSelectionToRestore = selectionData.documentSelection;
-        }
+        this.canApplySelectionToDocument = selectionData.documentSelectionIsInEditable;
```

Comment:
I would move this to inside the `restore` method.
Just in case someone calls `preserveSelection` but doesn't call `restore`, or calls its asynchronously some time later.

Reply:
Actually I would not use this flag mechanism, and, in case the document selection is not in editable, instead of calling `setSelection` I would call a `updateActiveSelection` method that does what you did in the `else` branch in `setSelection`. Of course you would have to refactor some of `setSelection`'s logic into smaller functions to avoid code duplication (like a `normalizeSelection` method).

Reply:
the `updateActiveSelection` name is already taken ^^
Some other name then...